test/test_coretasks.py

```diff
@@ -0,0 +1,119 @@
+# coding=utf-8
+"""coretasks.py tests"""
+from __future__ import unicode_literals, absolute_import, print_function, division
+
+import pytest
+
+from sopel import coretasks
+from sopel.module import VOICE, HALFOP, OP, ADMIN, OWNER
+from sopel.tools import Identifier
+from sopel.test_tools import MockSopel, MockSopelWrapper
+from sopel.trigger import PreTrigger, Trigger
+
+
+@pytest.fixture
+def sopel():
+    bot = MockSopel("Sopel")
+    return bot
+
+
+@pytest.fixture
+def sopel_bot(sopel):
+    pretrigger = PreTrigger(Identifier("Foo"), "PING abc")
+    bot = MockSopelWrapper(sopel, pretrigger)
+    bot.privileges = dict()
+    bot.users = dict()
+    return bot
+
+
+def test_bot_mixed_modes(sopel_bot):
+    """
+    Ensure mixed modes like +vha are tracked correctly.
+    Sopel 6.6.6 and older would assign all modes to all users. #1575
+    """
+
+    # RPL_NAMREPLY to create Users and (zeroed) privs
+    for user in set("Unothing Uvoice Uhalfop Uop Uadmin Uowner".split(" ")):
+        pretrigger = PreTrigger(
+            "Foo", ":test.example.com 353 Foo = #test :Foo %s" % user
+        )
+        trigger = Trigger(sopel_bot.config, pretrigger, None)
+        coretasks.handle_names(sopel_bot, trigger)
+
+    pretrigger = PreTrigger("Foo", "MODE #test +qvhao Uowner Uvoice Uhalfop Uadmin Uop")
+    trigger = Trigger(sopel_bot.config, pretrigger, None)
+    coretasks.track_modes(sopel_bot, trigger)
```

Comment:
```suggestion
    sopel_bot = MockSopelWrapper(sopel, trigger)
    coretasks.track_modes(sopel_bot, trigger)
```

So you can use it later.